### PR TITLE
Refactor presenters to use SimpleDelegator

### PR DESCRIPTION
**Pull request by fofr**
- Simplify the presenters
- Use delegation to access application_choice and application_form models to DRY up code

### CODE REVIEW

File: app/presenters/provider_interface/application_choice_presenter.rb [hunk 1]
```diff
@@ -1,24 +1,11 @@
...
-    end
-
-    def status
-      application_choice.status
-    end
-
+  class ApplicationChoicePresenter < SimpleDelegator
     def status_tag_text
-      application_choice.status.humanize.titleize
+      status.humanize.titleize
```
Comment (tijmenb) [inline app/presenters/provider_interface/application_choice_presenter.rb]: I'm not following how this works, how does it know where `status` is defined?

Comment (fofr) [reply app/presenters/provider_interface/application_choice_presenter.rb]: It's inherited from `< SimpleDelegator`, it causes method missing to resolve everything to the object given to the constructor - @duncanjbrown

**Comment by tijmenb** [comment]
I'd like to propose a different refactor that should achieve much the same:

https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/compare/alternative-refactor

This way we only delegate the methods that we need to sub-objects. The methods can stay private, and we don't have to make modifications to core models.

Thoughts?

**Comment by duncanjbrown** [comment]
We're going to go at this outside-in (https://trello.com/c/lf08PU3R/1233-provider-can-view-an-application) preferring the @tijmenb approach and enhancing the ApplicationChoicePresenter as we go. If we end up with lots of duplication we can look again

**Pull request closed without merging**